conda/common/configuration.py

```diff
@@ -1101,6 +1156,110 @@ def load(self, name, match):
             self._element_type,
             match.keyflag(),
             match.valueflags(self._element_type),
+            validation=self._validation,
+        )
+
+
+#: These are the allowable types for a UnionSequenceParameter
+UnionElementTypes = Union[MapParameter, SequenceParameter, PrimitiveParameter]
+
+
+class UnionSequenceParameter(Parameter):
```

Comment:
What about abstracting the existing `SequenceParameter`? One `element_type` is homogenous, multiple is heterogeneous.

Reply:
This could definitely be done with what I have right now. I will investigate. It's kind of weird to have two classes which essentially handle the same thing.

Reply:
@kenodegard, thanks for this suggestion. I have performed a refactor, and we now only have a single `SequenceParameter` class.